src/lib/userStorage/UserProfileStorage.js

```diff
@@ -9,17 +23,238 @@ export type ProfileField = {
 }
 
 export interface ProfileDB {
-  updateField(name: string, field: ProfileField): Promise<void>;
-  getProfileBy(field: String, value: string): Promise<void>;
+  user: App;
+}
+type ACK = {
+  ok: number,
+  err: string,
 }
 
-// export interface ProfileStorage {
-//     setProfileField
-// }
+export class UserProfileStorage {
+  constructor(wallet: GoodWallet, profiledb: ProfileDB) {
+    // const seed = Uint8Array.from(Buffer.from(pkeySeed, 'hex'))
+    // this.privateKey = TextileCrypto.PrivateKey.fromRawEd25519Seed(seed)
+    this.wallet = wallet
+    this.profiledb = profiledb
+  }
+
+  profileSettings: {} = {
+    fullName: { defaultPrivacy: 'public' },
+    email: { defaultPrivacy: 'private' },
+    mobile: { defaultPrivacy: 'private' },
+    mnemonic: { defaultPrivacy: 'private' },
+    avatar: { defaultPrivacy: 'public' },
+    smallAvatar: { defaultPrivacy: 'public' },
+    walletAddress: { defaultPrivacy: 'public' },
+    username: { defaultPrivacy: 'public' },
+  }
+
+  static indexableFields = {
+    email: true,
+    mobile: true,
+    mnemonic: true,
+    phone: true,
+    walletAddress: true,
+    username: true,
+  }
+
+  subscribersProfileUpdates = []
+
+  walletAddressIndex = {}
+
+  static cleanHashedFieldForIndex = (field: string, value: string): string => {
+    if (value === undefined) {
+      return value
+    }
+    if (field === 'mobile' || field === 'phone') {
+      return sha3(value.replace(/[_-\s]+/g, ''))
+    }
+    return sha3(`${value}`.toLowerCase())
+  }
+
+  static isValidValue(field: string, value: string, trusted: boolean = false) {
+    const cleanValue = UserProfileStorage.cleanHashedFieldForIndex(field, value)
+
+    if (!cleanValue) {
+      logger.warn(
+        `indexProfileField - field ${field} value is empty (value: ${value})`,
+        cleanValue,
+        new Error('isValidValue failed'),
+        { category: ExceptionCategory.Human },
+      )
+      return false
+    }
+
+    return true
+  }
+
+  _getProfileFields = profile => keys(profile).filter(field => !['_', 'initialized'].includes(field))
+
+  //TODO:  make sure profile contains walletaddress or enforce it in schema in realmdb
+  setProfile(profile): Promise<any> {
+    this.profiledb.Profiles.updateOne(
```

Comment:
here you use the setProfile from the ProfileDB interface
userstorage should be oblivious to the specific implementation of profiledb.
in this case you are making userstorage coupled with mongo db